staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -386,9 +386,14 @@ func IsDeleted(info *resource.Info, o *WaitOptions) (runtime.Object, bool, error
 		return false, nil
 	}
 
+	intrCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
 	intr := interrupt.New(nil, cancel)
 	err := intr.Run(func() error {
-		_, err := watchtools.UntilWithSync(ctx, lw, &unstructured.Unstructured{}, preconditionFunc, Wait{errOut: o.ErrOut}.IsDeleted)
+		_, err := watchtools.UntilWithSync(intrCtx, lw, &unstructured.Unstructured{}, preconditionFunc, Wait{errOut: o.ErrOut}.IsDeleted)
+		if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
I assume this will only work when https://github.com/kubernetes/kubernetes/pull/114578 lands?

Reply:
unfortunately, yes, it is correct